wolfcrypt/src/asn.c

```diff
@@ -16546,115 +16547,120 @@ static int DecodeSingleResponse(byte* source,
 
     prevIndex = idx;
 
-    /* When making a request, we only request one status on one certificate
-     * at a time. There should only be one SingleResponse */
+    /* wolfSSL only requests one status for one certificate at a time but
+       some OCSP responders can reply with multiple SingleResponse items.
+       Expect to handle one SingleResponse. Otherwise, we can process the
+       responses but only the last entry in the list is verified. */
 
-    /* Wrapper around the Single Response */
-    if (GetSequence(source, &idx, &length, size) < 0)
-        return ASN_PARSE_E;
+    while ((idx-prevIndex) < (word32)wrapperSz) {
```

Comment:
This while loop is the only addition here (rest is indent). While it properly iterates the responses it's only capturing the last "singleResponse" into `resp`. Either you need to filter based on the request data or you'll need to capture a list of these to compare later. Specifically `issuerHash`, `issuerKeyHash`, `serial`, `serialSz`, `status`, `thisDate`, `nextDate`.

In the customer report it looks like the responses are all for the same key, but different serial numbers.

Reply:
@dgarske. You are spot on! expanded to process the OCSP responses only and caveats were documented as a comment.